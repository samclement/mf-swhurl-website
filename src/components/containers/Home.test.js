/* eslint-env jest */

import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import Index from './Home.js'
// Nextjs routing support
import Router from 'next/router'
const mockedRouter = { push: () => {} }
Router.router = mockedRouter

describe('With Snapshot Testing', () => {
  it('Index page shows "mf.swhurl.com" H1', () => {
    const { asFragment } = render(<Index />)
    expect(asFragment()).toMatchSnapshot()
  })
})
