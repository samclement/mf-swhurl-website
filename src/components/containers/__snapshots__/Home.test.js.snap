// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Index page shows "mf.swhurl.com" H1 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c1 {
  text-align: center;
  text-transform: uppercase;
  width: 33%;
}

.c3 {
  padding: 0 10px;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
  font-weight: normal;
}

<div>
    <nav>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <a
            class="c2"
            href="/mens"
          >
            Men
          </a>
        </div>
        <div
          class="c3"
        >
          <h1>
            <a
              class="c2"
              href="/"
            >
              MATCHESFASHION
            </a>
          </h1>
        </div>
        <div
          class="c1"
        >
          <a
            class="c2"
            href="/womens"
          >
            Women
          </a>
        </div>
      </div>
    </nav>
    <h2>
      mf.swhurl.com
    </h2>
    <p>
      Example next.js app, including:
    </p>
    <ul>
      <li>
        nextjs (server-side rendering single-page application)
      </li>
      <li>
        styled-components
      </li>
      <li>
        linting with eslint
      </li>
      <li>
        jest snapshot testing
      </li>
      <li>
        prettier formatting
      </li>
      <li>
        lighthouse performance testing
      </li>
      <li>
        chrome browser testing (wdio)
      </li>
      <li>
        greenkeeper (automatic PRs for upgrading dependencies)
      </li>
      <li>
        kustomize (k8s deployment configuration)
      </li>
    </ul>
    <p>
      The application is deployed to a k8s cluster that includes:
    </p>
    <ul>
      <li>
        cert-manager (automatic TLS certificate provisioning from LetsEncript)
      </li>
      <li>
        oauth2-proxy (authentication against github)
      </li>
      <li>
        prometheus (infrastructure and applicaiton monitoring)
      </li>
      <li>
        fluentbit / loggly integration (log aggregation)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
